modules/system/console/CreateMigration.php

```diff
@@ -157,11 +158,13 @@ protected function prepareVars(): array
         }
 
         if ($this->option('create') && $this->option('update')) {
-            $this->error('The create & update options cannot both be set at the same time');
-            return false;
+            throw new InvalidArgumentException('The create & update options cannot both be set at the same time');
         }
 
         if ($this->option('create')) {
+            if (empty($model)) {
+                throw new InvalidArgumentException('The create options requires the --model option');
+            }
```

Comment:
This shouldn't be required, you can have a creation migration without a model so long as you have a table name, and that is checked in line 176. 

The auto scaffolding of table structure based on model fields should be a separate flag that can work for create or update contexts alike.

Reply:
> This shouldn't be required, you can have a creation migration without a model so long as you have a table name, and that is checked in line 176.

It currently doesn't extract the model from the table name... the model was not used in the basic migration prior to this.

> The auto scaffolding of table structure based on model fields should be a separate flag that can work for create or update contexts alike.

Updating fields is a whole different story, this PR doesn't addess this.

Reply:
Right, but I'm saying that this check shouldn't be targeting the `--create` option, that's overloading the option to mean both create stub and autocreate from model stub, when instead it should really be a separate option altogether with its own validation to handle the autogeneration from the model.

We don't have to be perfect, but in theory we could apply the same logic for an update context, it would just be up to the developer to edit the results.

Reply:
> Right, but I'm saying that this check shouldn't be targeting the `--create` option, that's overloading the option to mean both create stub and autocreate from model stub, when instead it should really be a separate option altogether with its own validation to handle the autogeneration from the model.

Ok, autocreate from model now only triggers if --model option is supplied.

> We don't have to be perfect, but in theory we could apply the same logic for an update context, it would just be up to the developer to edit the results.

Ok, the fields are now included in the update stub if --update and --model options are used.